### PR TITLE
[Site creation] Style SiteSegmentsWizardContent

### LINKED ISSUE #10483: [Site creation] Style SiteSegmentsWizardContent
- [x] Fix content insets (the header is under the navigation bar
- [x] Apply styles to cells. Look into reusing the cells in MySites

### PR DESCRIPTION
**Pull request by ctarda**
Fixes #10483 

A round of polishing to SiteSegmentsWizardContent by:
- [ ] Update the cells to match the styles in Zeplin. The second line of the cell (subtitle) should overflow to two lines if the content is large enough. Title and subtitle, as a block, should be centred vertically. The icon should be smaller.
- [ ] Prevent the header from being hidden by the top navigation bar

Before:
simulator screen shot - iphone x - 2018-11-19 at 13 56 57

After:
simulator screen shot - iphone x - 2018-11-19 at 14 12 17

Please note: 
- the table view header still overflows over the table view. This will be fixed in an upcoming PR (story #10479)
- as part of the diff, there is an update to the mock data (`SiteSegmentsService`). This service, like the rest of mocks will be removed before release

To test:
- Checkout the branch, run the tests.
- Navigate to My Sites > Tap + > Enhanced site creation

### CODE REVIEW

File: WordPress/Classes/Services/SiteSegmentsService.swift [hunk 1]
```diff
@@ -19,21 +19,29 @@ final class MockSiteSegmentsService: SiteSegmentsService {
...
         let identifier = Identifier(value: id)
         return SiteSegment(identifier: identifier,
-                           title: "Mock",
-                           subtitle: "Mock subtitle",
+                           title: "Blogger",
+                           subtitle: "Publish a collection of posts",
+                           icon: URL(string: "https://s.w.org/style/images/about/WordPress-logotype-standard.png")!)
+    }
+
+    private func longSubtitle(id: String) -> SiteSegment {
```
Comment (kwonye) [inline WordPress/Classes/Services/SiteSegmentsService.swift]: I like the addition of this for testing! 👍

**Comment by ctarda** [comment]
Oops, good catch @kwonye thanks!

I've pushed an update to bring the branch up to date with `develop` and to tint the icon. The icon of the Segment with a two line subtitle should be blue, while the others should be red, as declared in `MockSiteSegmentsService`

simulator screen shot - iphone x - 2018-11-20 at 11 27 46

**Comment by kwonye** [comment]
Looks good!

**Comment by SylvesterWilmott** [comment]
Looks good!

For each cell in the table there is a title and subtitle. It appears the title is larger than 17pt?

**Comment by kwonye** [comment]
@SylvesterWilmott I'll have you be the final arbiter of whether we can merge this or not 👍

**Comment by SylvesterWilmott** [comment]
Also just noticed, the subtitle text appears to be 12pt? In this case it should be 15pt.

You can find details about text sizes in zeplin by referencing the right hand sidebar while a text layer is selected.

**Comment by ctarda** [comment]
Thanks @SylvesterWilmott . As with the previous PR I tried to approximate as much as possible. 

The Title is now styled with `body` as style and `semibold` as weight. The Subtitle is `callout` and `regular`

**Comment by ctarda** [comment]
And I forgot to add a screenshot:

simulator screen shot - iphone x - 2018-11-21 at 14 05 37

**Comment by SylvesterWilmott** [comment]
Thanks!

Text sizes look good

- Can we allow those table cells to expand with the text. The default state would be one line for the title and one line for the subtitle but as we add more lines the cells can change height:

screenshot 2018-11-21 at 08 54 10

- Can we change the text color in the cells to `darkGrey`

**Comment by ctarda** [comment]
Thanks @SylvesterWilmott I think this is ready for another round. 

I made the cells support dynamic height. I double checked the test color and it is darkGrey, and it seems to match the RGB values declared on Zeplin. Please let me know if that's not the case, to take another look.

This is how this looks on an iPhone SE:
simulator screen shot - iphone se - 2018-11-26 at 09 41 46

This would be an iPhone X:
simulator screen shot - iphone x - 2018-11-26 at 09 40 19

And this would be on the new iPad Pro 11"
simulator screen shot - ipad pro 11-inch - 2018-11-26 at 09 42 59

@kwonye You might want to take a look at the last couple of commits, as I made some slight changes to the constraints attached to the header (I changed the width from being equal to the table view's to be lessThanOrEqualTo the table view's)

**Comment by SylvesterWilmott** [comment]
Looks good!

- For the text it still appears as black in the screenshots
screenshot 2018-11-26 at 09 16 53

- As mentioned in 10545: the tablet max-widths of the table should be 512pt

**Comment by ctarda** [comment]
Thanks @SylvesterWilmott 

I've updated the width constraints for tablets (regular height and width). Here is how it looks on a 11" iPad Pro now (I have not scaled the screenshot):

simulator screen shot - ipad pro 11-inch - 2018-11-27 at 12 37 23

I've double and triple checked the screenshot with Pixelmator and it seems like the color of the cells title and subtitle seems to be R: 46, G: 68, B: 83. Would you mind checking that again please?

screenshot 2018-11-27 at 12 46 23 pm

**Comment by SylvesterWilmott** [comment]
Thanks!

Layout looks great, colours are good.

Couple more minor things:

- The table row strokes should be `greyLighten20`
- Can we remove the cell stroke from the last child of the list?

**Comment by ctarda** [comment]
Thanks!

I've updated the separator color and removed the one in the last cell.

simulator screen shot - iphone x - 2018-11-27 at 18 57 55

**Comment by SylvesterWilmott** [comment]
One final thing (I didn't notice before) can we tint the chevron also to `greyLighten20`.

After that I think we're all good

**Comment by ctarda** [comment]
Thanks @SylvesterWilmott I've updated the chevron.

simulator screen shot - iphone x - 2018-11-28 at 09 29 52

My checklist for the upcoming UI reviews would be:

- [ ] Width of lists on iPad
- [ ] Fonts and colours in cells
- [ ] Color of separator
- [ ] Hide separator for last cell
- [ ] Color of disclosure indicators

Thanks!

**Comment by SylvesterWilmott** [comment]
Hmm the colour change of the separators didn't seem to make an effect. Otherwise everything looks good!

**Comment by ctarda** [comment]
Sorry about that, it seems like I pushed the change to the wrong branch. I've updated it again.

simulator screen shot - iphone x - 2018-11-28 at 17 32 46

**Pull request merged**